src/main/java/org/scijava/log/AbstractLogService.java

```diff
@@ -107,7 +125,7 @@ public void removeLogListener(final LogListener l) {
 	}
 
 	@Override
-	public void notifyListeners(final int level, final Object msg,
+	public void notifyListeners(final LogLevel level, final Object msg,
```

Comment:
And there is another reason not to introduce `LogLevel`: it breaks backwards API compatibility. The changes done earlier in this topic branch were done very carefully so that existing consumers of `LogService` will actually keep working without recompilation—at least in theory. :wink: If you want to introduce `LogLevel`, we need to keep _all_ the existing signatures that use `int level` also, deprecate them, and have them delegate to the new signatures. Otherwise, this change needs to target SJC 3.0.0, which is the next major i.e. breaking release.